persistence/internal/providertest/offsetstore/transaction.go

```diff
@@ -0,0 +1,198 @@
+package offsetstore
+
+import (
+	"sync"
+
+	"github.com/dogmatiq/infix/persistence"
+	"github.com/dogmatiq/infix/persistence/internal/providertest/common"
+	"github.com/dogmatiq/infix/persistence/subsystem/offsetstore"
+	"github.com/onsi/ginkgo"
+	"github.com/onsi/ginkgo/extensions/table"
+	"github.com/onsi/gomega"
+)
+
+// DeclareTransactionTests declares a functional test-suite for a specific
+// offsetstore.Transaction implementation.
+func DeclareTransactionTests(tc *common.TestContext) {
+	ginkgo.Describe("type offsetstore.Transaction", func() {
+		var (
+			dataStore  persistence.DataStore
+			repository offsetstore.Repository
+			tearDown   func()
+		)
+
+		ginkgo.BeforeEach(func() {
+			dataStore, tearDown = tc.SetupDataStore()
+			repository = dataStore.OffsetStoreRepository()
+		})
+
+		ginkgo.AfterEach(func() {
+			tearDown()
+		})
+
+		ginkgo.Describe("func SaveOffset()", func() {
+			ginkgo.When("application has no previous offsets associated", func() {
+				ginkgo.It("saves an offset with the current offset equal to zero", func() {
```

Comment:
```suggestion
				ginkgo.It("saves the offset", func() {
```

This test doesn't need a lengthy description, it's just the common case, and will make sense when you see it next to the OCC test (see comment below).